### PR TITLE
Feature: Auth

### PR DESCRIPTION
**Pull request by iamclaytonray**

### CODE REVIEW

File: src/core/user/index.ts [hunk 1]
```diff
@@ -1,12 +1,5 @@
 // Entity
-export { User } from '../../entity/UserEntity';
+export { User, Role } from '../../entity/UserEntity';
```
Comment (mateodelnorte) [inline src/core/user/index.ts]: Do we get much benefit from having this extra file re-export these items? It's one less file and fewer lines of code if the files that use User, Role, and the items from UserController import them directly. The counter argument would be if the items within those files are likely to be moved to other files in the future, and the external interface should be maintained. Is that what we're thinking?

**Comment by mateodelnorte** [comment]
Do we want to add some tests here?

**Comment by iamclaytonray** [comment]
> Do we want to add some tests here?

I started writing some E2E tests so I'll finish up on those

**Comment by mateodelnorte** [comment]
Getting the following when trying to create a user
```
darkstore-server_1  | 🛑 secretOrPrivateKey must have a value
```
I can't find secretOrPrivateKey in the codebase. Am I missing an env var?

**Comment by iamclaytonray** [comment]
> I can't find secretOrPrivateKey in the codebase. Am I missing an env var?

Yes: JWT_SECRET=something

**Comment by mateodelnorte** [comment]
Looking good. Able to create a user, redirected to account page.

**Pull request merged**